server/map_tree_updater.go

```diff
@@ -38,120 +38,152 @@ type mapTreeUpdater struct {
 
 // update updates the sparse Merkle tree at the passed-in revision with the
 // given leaf updates, and writes it to the storage. Returns the new root hash.
-func (t *mapTreeUpdater) update(ctx context.Context, tx storage.MapTreeTX, hkv []merkle.HashKeyValue, rev int64) ([]byte, error) {
-	// Work around a performance issue when using the map in
-	// single-transaction mode by preloading all the nodes we know the
-	// sparse Merkle writer is going to need.
+func (t *mapTreeUpdater) update(ctx context.Context, tx storage.MapTreeTX, upd []smt.NodeUpdate, writeRev int64) ([]byte, error) {
+	// TODO(pavelkalinnikov): Return an error when no updates are given.
```

Comment:
Another option is to move this [annoying] clause to `map_rpc_server.go`. The closer it is to the API level the better.

Reply:
Done.